ethereum_history_api/circuits/lib/src/merkle_patricia_proofs/proof.nr

```diff
@@ -23,6 +25,41 @@ pub fn leaf_get_len<MAX_LEN>(leaf: Leaf<MAX_LEN>) -> u32 {
     leaf_unpadded_len as u32
 }
 
+// Prefixes mean:
+// 0 - Extension node, even number of nibbles
+// 1 - Extension node, odd number of nibbles
+// 2 - Leaf node, even number of nibbles
+// 3 - Leaf node, odd number of nibbles
+pub fn parity(prefix: u8) -> u8 {
+    assert(prefix < 4, "Prefix must be 0, 1, 2 or 3");
+    let result = prefix % 2;
+    assert((result == EVEN) | (result == ODD), "Invalid parity");
+
+    result
+}
+
+// Leaf and extension nodes are RLP lists with 2 elements: prefixed key nibbles (in bytes) and value.
+// This function removes the prefix and returns appropriate number of key nibbles according to the parity of the prefix (see function: parity).
+pub fn strip_prefix<MAX_KEY_LEN, MAX_KEY_NIBBLE_LEN>(
+    prefixed_key_bytes: [u8; MAX_KEY_LEN],
+    prefixed_len: u64
+) -> ([u8; MAX_KEY_NIBBLE_LEN], u64) {
+    assert(MAX_KEY_LEN * 2 == MAX_KEY_NIBBLE_LEN);
```

Comment:
Why do we have to make this assert? Are we always using these  global values of max lengths? If not can't there be a case when we want the result array to be not padded?

Reply:
I think we'll always pass padded values and we have to have some limits for `MAX_KEY_LEN` and the most sensible one is that `MAX_KEY_NIBBLE_LEN == 2 * MAX_KEY_LEN`. What do you think @LogvinovLeon?

Reply:
Add assertion message